src/dos/dossb.c

```diff
@@ -0,0 +1,550 @@
+/* Sound Blaster I/O routines, common for SB8, SBPro, and SB16,
+ * from libMikMod. Written by Andrew Zabolotny <bit@eltech.ru>
+ * Further bug fixes by O. Sezer <sezero@users.sourceforge.net>
+ *
+ * Extended Module Player
+ * Copyright (C) 1996-2021 Claudio Matsuoka and Hipolito Carraro Jr
+ *
+ * This file is part of the Extended Module Player and is distributed
+ * under the terms of the GNU General Public License. See the COPYING
+ * file for more information.
+ */
+
+#include <stdlib.h>
+#include <dpmi.h>
+#include <go32.h>
+#include <dos.h>
+#include <sys/nearptr.h>
+#include <sys/farptr.h>
+#include <string.h>
+
+#include "dossb.h"
+
+/********************************************* Private variables/routines *****/
+
+__sb_state sb;
+
+/* Wait for SoundBlaster for some time */
+#if !defined(__GNUC__) || (__GNUC__ < 3) || (__GNUC__ == 3 && __GNUC_MINOR__ == 0)
+# define _func_noinline volatile /* match original code */
+# define _func_noclone
+#else
+/* avoid warnings from newer gcc:
+ * "function definition has qualified void return type" and
+ * function return types not compatible due to 'volatile' */
+# define _func_noinline __attribute__((__noinline__))
+# if (__GNUC__ < 4) || (__GNUC__ == 4 && __GNUC_MINOR__ < 5)
+#  define _func_noclone
+# else
+#  define _func_noclone __attribute__((__noclone__))
+# endif
+#endif
+_func_noinline
```

Comment:
Since `_func_noinline` can be defined as `volatile`, the function prototype in `dossb.h` should have these attributes as well.

Reply:
OK, will change it soon.